addons/sale/models/payment_transaction.py

```diff
@@ -98,17 +101,43 @@ def _reconcile_after_done(self):
         self._invoice_sale_orders()
         res = super()._reconcile_after_done()
         if self.env['ir.config_parameter'].sudo().get_param('sale.automatic_invoice') and any(so.state in ('sale', 'done') for so in self.sale_order_ids):
-            default_template = self.env['ir.config_parameter'].sudo().get_param('sale.default_invoice_email_template')
-            if default_template:
-                for trans in self.filtered(lambda t: t.sale_order_ids.filtered(lambda so: so.state in ('sale', 'done'))):
-                    trans = trans.with_company(trans.acquirer_id.company_id).with_context(
-                        mark_invoice_as_sent=True,
-                        company_id=trans.acquirer_id.company_id.id,
-                    )
-                    for invoice in trans.invoice_ids.with_user(SUPERUSER_ID):
-                        invoice.message_post_with_template(int(default_template), email_layout_xmlid="mail.mail_notification_paynow")
+            self.filtered(lambda t: t.sale_order_ids.filtered(lambda so: so.state in ('sale', 'done')))._send_invoice()
         return res
 
+    def _send_invoice(self):
+        default_template = self.env['ir.config_parameter'].sudo().get_param('sale.default_invoice_email_template')
+        if not default_template:
+            return
+
+        for trans in self:
+            trans = trans.with_company(trans.acquirer_id.company_id).with_context(
+                company_id=trans.acquirer_id.company_id.id,
+            )
+            invoice_to_send = trans.mapped('invoice_ids').filtered(lambda i: not i.is_move_sent and i.state == 'posted' and i._is_ready_to_be_sent())
+            invoice_to_send.is_move_sent = True # Mark invoice as sent
+            for invoice in invoice_to_send.with_user(SUPERUSER_ID):
+                invoice.message_post_with_template(int(default_template), email_layout_xmlid="mail.mail_notification_paynow")
+
+    def _cron_send_invoice(self):
+        """
+            Cron to send invoice that where not ready to be send directly after posting
+        """
+        if not self.env['ir.config_parameter'].sudo().get_param('sale.automatic_invoice'):
+            return
+
+        # No need to retreive old transaction
+        retry_limit_date = datetime.now() - relativedelta.relativedelta(days=2)
+        # Retrieve all transactions matching the criteria for post-processing
+        self.search([
+            ('state', '=', 'done'),
+            ('is_post_processed', '=', True),
+            ('invoice_ids.is_move_sent', '=', False),
+            ('invoice_ids.state', '=', 'posted'),
```

Comment:
```suggestion
            ('invoice_ids', 'in', self.env['account.move']._search([
                ('is_move_sent', '=', False),
                ('state', '=', 'posted'),
            ]),
```
might be more efficient with a subquery, to test :).

Reply:
Both query are quite fast the query plan of your suggestion is a bit lighter